mongo/cursor.go

```diff
@@ -246,7 +246,10 @@ func (c *Cursor) All(ctx context.Context, results interface{}) error {
 	var index int
 	var err error
 
-	defer c.Close(ctx)
+	// Defer a call to Close to try to clean up the cursor server-side when all
+	// documents have been exhausted. Use context.Background() to ensure Close
```

Comment:
```suggestion
	// documents have not been exhausted. Use context.Background() to ensure Close
```

IIUC when all documents are exhausted, the cursor ID returned by the server is 0, and `killCursors` does not need to be sent.